### PR TITLE
Add a validator for ELB names

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
The LoadBalancerName has some constraints that we can validate for ahead of time:

> This name must be unique within your set of load balancers for the region, must have a maximum of 32 characters, must contain only alphanumeric characters or hyphens, and cannot begin or end with a hyphen.

From: http://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_CreateLoadBalancer.html

My current implementation of the regex doesn't quite work.  A one character name ought to work but I have created a regex that will fail on names of one character.  I could use some help on it.

### CODE REVIEW

File: tests/test_validators.py [hunk 1]
```diff
@@ -73,6 +73,14 @@ def test_s3_bucket_name(self):
             with self.assertRaises(ValueError):
                 s3_bucket_name(b)
 
+    def test_elb_name(self):
+        for b in ['a', 'a'*32, 'wick3d-elb-name', 'Wick3d-ELB-Name']:
```
Comment (markpeek) [inline tests/test_validators.py]: Given the regex you might want to add:
```
a-a
aa
aaa
```

File: tests/test_validators.py [hunk 2]
```diff
@@ -73,6 +73,14 @@ def test_s3_bucket_name(self):
...
+            elb_name(b)
+        for b in ['a'*63, 'invalid_elb', '-invalid-elb',
```
Comment (markpeek) [inline tests/test_validators.py]: Probably want 'a'*33 instead of 63. Add 'a-' as well.

File: troposphere/validators.py [hunk 3]
```diff
@@ -73,6 +73,14 @@ def s3_bucket_name(b):
         raise ValueError("%s is not a valid s3 bucket name" % b)
 
 
+def elb_name(b):
+    elb_name_re = compile(r'^[a-zA-Z0-9][a-zA-Z0-9\-]{0,30}[a-zA-Z0-9]{1}$')
```
Comment (markpeek) [inline troposphere/validators.py]: Try something like this:
```
^[a-zA-Z0-9](?:[a-zA-Z0-9\-]{0,30}[a-zA-Z0-9]{1})?$
```

File: tests/test_validators.py (hunk 1 above)
Comment (chrisgilmerproj) [inline tests/test_validators.py]: Added in 9b55342afe7a2f14cb126f49441fb19a8d514470

File: tests/test_validators.py (hunk 2 above)
Comment (chrisgilmerproj) [inline tests/test_validators.py]: Good catch.  Copy pasta.  Fixed and expanded in 9b55342afe7a2f14cb126f49441fb19a8d514470

File: troposphere/validators.py (hunk 3 above)
Comment (chrisgilmerproj) [inline troposphere/validators.py]: This worked! Nice call.  I've added it here: 9b55342afe7a2f14cb126f49441fb19a8d514470

I tested manually a bit on http://pythex.org/ just for context.

**Comment by chrisgilmerproj** [comment]
@markpeek - Thanks for the help on this.  It looks to have passed the tests.  Do you want me to squash the commits before you accept it?

**Comment by markpeek** [comment]
@chrisgilmerproj thanks for making the changes. I'll have time to review further and commit later. I'll squash as part of the github commit flow.

**Comment by chrisgilmerproj** [comment]
I appreciate it.  Thanks so much.

**Comment by markpeek** [comment]
Thanks!

**Pull request merged**